src/v/iceberg/BUILD

```diff
@@ -184,3 +184,24 @@ redpanda_cc_library(
         "//src/v/json",
     ],
 )
+
+redpanda_cc_library(
+    name = "values",
+    srcs = [
+        "values.cc",
+    ],
+    hdrs = [
+        "values.h",
+    ],
+    include_prefix = "iceberg",
+    deps = [
+        "//src/v/bytes:hash",
+        "//src/v/bytes:iobuf",
+        "//src/v/bytes:iobuf_parser",
```

Comment:
looks like at least bytes/hash and iobuf_parser can be implementation deps
```
implementation_deps = [
        "//src/v/bytes:hash",
        "//src/v/bytes:iobuf_parser",
],
```

Reply:
Ah cool. Is a good heuristic for whether something should be an implementation_dep to ask myself whether a library is only included by a cc file?

Reply:
Done

Reply:
> Ah cool. Is a good heuristic for whether something should be an implementation_dep to ask myself whether a library is only included by a cc file?

We chatted about this and the answer is, looking at the cc files is a fine heuristic but not necessarily the only one. There are other considerations, like if the .cc file is entirely used by this library and shouldn't be a part of the public interface.